src/gluonts/dataset/pandas.py

```diff
@@ -84,67 +84,66 @@ class PandasDataset:
     timestamp: Optional[str] = None
     freq: Optional[str] = None
     static_features: pd.DataFrame = pd.DataFrame()
-    ignore_last_n_targets: int = 0
+    future_length: int = 0
     unchecked: bool = False
     assume_sorted: bool = False
     dtype: Type = np.float32
-    _pairs: Iterable[Tuple[Any, Union[pd.Series, pd.DataFrame]]] = field(
-        init=False
-    )
     _static_reals: pd.DataFrame = field(init=False)
     _static_cats: pd.DataFrame = field(init=False)
 
     def __post_init__(self):
         if isinstance(self.dataframes, dict):
-            self._pairs = self.dataframes.items()
+            pairs = self.dataframes.items()
         elif isinstance(self.dataframes, (pd.Series, pd.DataFrame)):
-            self._pairs = [pair_with_item_id(self.dataframes)]
+            pairs = [(None, self.dataframes)]
         else:
             assert isinstance(self.dataframes, SizedIterable)
-            self._pairs = Map(pair_with_item_id, self.dataframes)
+            pairs = Map(pair_with_item_id, self.dataframes)
 
-        assert isinstance(self._pairs, SizedIterable)
+        assert isinstance(pairs, SizedIterable)
+        self._data_entries = Map(self._pair_to_dataentry, pairs)
+
+        if self.feat_dynamic_real is None:
+            self.feat_dynamic_real = pd.DataFrame()
+
+        if self.past_feat_dynamic_real is None:
+            self.past_feat_dynamic_real = pd.DataFrame()
```

Comment:
Does the typing of these attributes need to be adjusted?

Reply:
The problem is that one can pass a `None`, but internally it is guaranteed to be not None.

Maybe it's easier to not make this a dataclass.

Reply:
But then I don't understand the change: this is supposed to be a list of column names, here it gets turned into a dataframe if it was None, what happens in the other case?

Reply:
I think `dataclass` makes sense in this case, I'm not yet ready to give that up